pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -323,28 +329,42 @@ def setDownsampling(self, ds=None, auto=None, method=None):
             if self.opts['downsample'] != ds:
                 changed = True
                 self.opts['downsample'] = ds
-                
+
         if auto is not None and self.opts['autoDownsample'] != auto:
             self.opts['autoDownsample'] = auto
             changed = True
-                
+
         if method is not None:
             if self.opts['downsampleMethod'] != method:
                 changed = True
                 self.opts['downsampleMethod'] = method
-        
+
         if changed:
             self.xDisp = self.yDisp = None
             self.updateItems()
-        
+
     def setClipToView(self, clip):
         if self.opts['clipToView'] == clip:
             return
         self.opts['clipToView'] = clip
         self.xDisp = self.yDisp = None
         self.updateItems()
-        
-        
+
+    def setDynamicRangeLimit(self, limit):
+        """
+        Limit the off-screen positions of data points at large magnification
+        This avoids errors with plots not displaying because their visibility is incorrectly determined. The default setting repositions far-off points to be within +-1E+06 times the viewport height.
+        =============== =============================================================
+        **Arguments:**
+        limit           (float or None) Maximum allowed vertical distance of plotted points in units of viewport height.
```

Comment:
wait, I thought limit was of type `bool`, if it can be `float`, then we should most definitely not be using the `is` operator

Reply:
unless explicitly checking for `None`